thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java

```diff
@@ -155,11 +175,44 @@ public AnomalyStatsApi computeAnomalyStats(final DaoFilter filter) {
         ? anomalyFeedbacksSupplier.get()
         : getAnomalyFeedbacks(filter);
     return new AnomalyStatsApi()
-        .setTotalCount(anomalyManager.countParentAnomalies(filter))
-        .setCountWithFeedback((long) allFeedbacks.size())
+        .setTotalCount(countTotal(filter))
+        .setCountWithFeedback(countFeedbacks(filter))
         .setFeedbackStats(aggregateFeedbackTypes(allFeedbacks));
   }
 
+  private Long countTotal(final DaoFilter filter) {
```

Comment:
The filter created internally in the method is to impose the constraints necessary for counting anomalies, such as filtering child and ignored anomalies. The filter which the method accepts is to provide additional filters on top of the basic filters such as anomaly startTime, endTime or is feedback present. Another reason to create a local filter is to avoid updating the passed filter from caller as it may be used at multiple places. Also allowing the caller function to not worry and pass null value if it wants just the raw count.